### ISSUE TITLE
`AdaptedModel` does not save weights

### ISSUE THREAD

**Issue by pplantinga** [speechbrain/speechbrain#2715]
### Describe the bug

The saved `state_dict` is always empty for `AdaptedModel` because the `state_dict` does not contain gradient information, it only has the actual parameter values.

```python
@checkpoints.mark_as_saver
    def saver(self, path):
        """Saves only the trainable parameters."""
        state_dict = {
            n: p for n, p in self.state_dict().items() if p.requires_grad
        }
        torch.save(state_dict, path)
```

`state_dict()` is not right here because it doesn't have the correct `requires_grad` value.

### Expected behaviour

Saving an `AdaptedModel` should save the adapted parameters.
